chain/indexer.go

```diff
@@ -297,9 +319,9 @@ func (t *TipSetIndexer) TipSet(ctx context.Context, ts *types.TipSet) error {
 				ll.Debugw("task data persisted", "task", task, "time", time.Since(start))
 			}(task, p)
 		}
-		wg.Wait()
 		ll.Debugw("tipset complete", "total_time", time.Since(start))
 	}()
+	wg.Wait()
```

Comment:
This will force the extraction and persistence to be locked in step, but they are supposed to be out of step. The idea is that we can be extracting a tipset while persisting the previous one (which is mostly IO wait).

Can you take this out and let's figure out a fix for #374 separately

Reply:
I have reverted this change here: 288beafa4673e74bbaf3fa024799254f06e61f1e
And applied a hacky-fix here: fb47362dbd11c562b63478b001f871691d0ef6f7

I agree this, nor the hack are the correct solution, but without one of them all vector tests fail. I believe this is because once the walker has indexed the last tipset in its walk it calls this method and exists regardless of this go routine completing its persistence work.

Reply:
:cry: